--- a/backport-include/net/genetlink.h
+++ b/backport-include/net/genetlink.h
@@ -262,6 +262,19 @@ static inline int genlmsg_parse(const struct nlmsghdr *nlh,
 	return __nlmsg_parse(nlh, family->hdrsize + GENL_HDRLEN, tb, maxtype,
 			     policy, NL_VALIDATE_STRICT, extack);
 }
+#elif LINUX_VERSION_IN_RANGE(6,6,59, 6,7,0) || \
+	LINUX_VERSION_IN_RANGE(6,1,115, 6,2,0) || \
+	LINUX_VERSION_IN_RANGE(5,15,170, 5,16,0) || \
+	LINUX_VERSION_IN_RANGE(5,10,229, 5,11,0) || \
+	LINUX_VERSION_IN_RANGE(5,4,285, 5,5,0)
+static inline
+int backport_genlmsg_multicast_allns(const struct genl_family *family,
+				     struct sk_buff *skb, u32 portid,
+				     unsigned int group, gfp_t flags)
+{
+	return genlmsg_multicast_allns(family, skb, portid, group);
+}
+#define genlmsg_multicast_allns LINUX_BACKPORT(genlmsg_multicast_allns)
 #endif /* LINUX_VERSION_IS_LESS(5,2,0) */
 
 #if LINUX_VERSION_IS_EQU(5,14,0) && RHEL_RELEASE_IS_GEQ(9,3)
